staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -106,6 +107,64 @@ func ParsePrivateKeyPEM(keyData []byte) (interface{}, error) {
 	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
 }
 
+// ParsePublicKeysPEM is a helper function for reading an array of rsa.PublicKey or ecdsa.PublicKey from a PEM-encoded byte array.
+// Reads public keys from both public and private key files.
+func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
+	ok := false
+	var keys []interface{}
+	for len(keyData) > 0 {
+		var block *pem.Block
+		block, keyData = pem.Decode(keyData)
+		if block == nil {
+			break
+		}
+		// Only use PEM "PUBLIC KEY" or "PRIVATE KEY" blocks without extra headers
+		if block.Type != PublicKeyBlockType || block.Type != PrivateKeyBlockType || len(block.Headers) != 0 {
```

Comment:
yeah, this looks to be unneeded, as we're just doing a switch based off what the block is. if it doesn't match, then we'll skip the block and on to the next